packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3727,6 +3727,12 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     _updateRemoteEditingValueIfNeeded();
     _startOrStopCursorTimerIfNeeded();
     _updateOrDisposeSelectionOverlayIfNeeded();
+    if (_hasFocus && !_value.selection.isValid) {
+      // If this field is focused and the selection is invalid, place the cursor at
+      // the end. Does not rely on _handleFocusChanged because it makes selection
+      // handles visible on Android.
+      widget.controller.selection = TextSelection.collapsed(offset: _value.text.length);
```

Comment:
This seems reasonable to me.  There have been some problems with invalid selection.  The engine often will set the selection to `-1, -1`, to mean "no selection" (and elsewhere, sometimes a `null` selection is used to mean no selection).  If the field is focused though, it makes sense to set the cursor to the end of the field as an improvement on those kinds of bugs.